StRoot/StFstRawHitMaker/StFstRawHitMaker.cxx

```diff
@@ -0,0 +1,598 @@
+	mPedVec(kFstNumElecIds, std::vector<float>(kFstNumTimeBins, 0)),
+	mTotRmsVec(kFstNumElecIds, std::vector<float>(kFstNumTimeBins, 0)),
+        mRanRmsVec(kFstNumElecIds, std::vector<float>(kFstNumTimeBins, 0)),
+	mGainVec(kFstNumElecIds, 0),
+	mMappingVec(kFstNumElecIds, 0),
+	mConfigVec(kFstNumApvs, 1),
+	mDataType(0)
+{
+}
+
+StFstRawHitMaker::~StFstRawHitMaker()
+{
+	delete mFstCollectionPtr; mFstCollectionPtr = 0;
+}
+
+
+/**
+ * Init(): prepare the FST raw hit collection
+ * in the dataset m_DataSet (data member of StMaker)
+ */
+Int_t StFstRawHitMaker::Init()
+{
+	ToWhiteConst("fstRawHitAndCluster", mFstCollectionPtr);
+
+	return kStOk;
+}
+
+
+/**
+ * InitRun(): access FST calibration DB and retrieve the calibration information
+ * from Db tables
+ */
+Int_t StFstRawHitMaker::InitRun(Int_t runnumber)
+{
+	Int_t ierr = kStOk;
+
+	TObjectSet *fstDbDataSet = (TObjectSet *)GetDataSet("fst_db");
+	StFstDb *mFstDb = NULL;
+
+	if (fstDbDataSet) {
+		mFstDb = (StFstDb *)fstDbDataSet->GetObject();
+		assert(mFstDb);
+	}
+	else {
+		LOG_ERROR << "InitRun : no fstDb" << endm;
+		return kStErr;
+	}
+
+	// FST control parameters
+	const fstControl_st *fstControlTable = mFstDb->getControl() ;
+
+	if (!fstControlTable)  {
+		LOG_ERROR << "Pointer to FST control table is null" << endm;
+		ierr = kStErr;
+	}
+	else {
+		mMinHitCut            = fstControlTable[0].kFstMinHitCutDefault;
+		mMedHitCut            = fstControlTable[0].kFstMedHitCutDefault;
+		mMaxHitCut            = fstControlTable[0].kFstMaxHitCutDefault;
+		mCmnCut               = fstControlTable[0].kFstCMNCutDefault;
+		mChanMinRmsNoiseLevel = fstControlTable[0].kFstChanMinRmsNoiseLevel;
+		mChanMaxRmsNoiseLevel = fstControlTable[0].kFstChanMaxRmsNoiseLevel;
+		mApvMaxCmNoiseLevel   = fstControlTable[0].kFstApvMaxCmNoiseLevel;
+		mALLdata              = fstControlTable[0].kFstAlldata;
+		mADCdata              = fstControlTable[0].kFstADCdata;
+		mZSdata               = fstControlTable[0].kFstZSdata;
+		mDefaultTimeBin       = fstControlTable[0].kFstDefaultTimeBin;
+		mCurrentTimeBinNum    = fstControlTable[0].kFstCurrentTimeBinNum;
+		mMinNumOfRawHits      = fstControlTable[0].kFstMinNumOfRawHits;
+		mMaxNumOfRawHits      = fstControlTable[0].kFstMaxNumOfRawHits;
+	}
+
+	// FST pedestal/rms table
+	const fstPedNoise_st *gPN = mFstDb->getPedNoise();
+
+	if ( !gPN ) {
+		LOG_ERROR << "Pointer to FST pedestal/noise table is null" << endm;
+		ierr = kStErr;
+	}
+	else {
+		for (int i = 0; i < kFstNumApvs; i++) {
+                     for ( int j = 0; j < kFstNumRStripsPerSensor; j++) {
+                          for ( int k = 0; k < kFstNumTimeBins; k++) {
+   			       LOG_DEBUG << Form(" Print entry %d-%d-%d : CM noise=%f ", i, j, k, (float)gPN[0].cmNoise[(i*kFstNumRStripsPerSensor+j)*kFstNumTimeBins+k] / 100.) << endm;
+			       mCmnVec[i][j][k] = (float)gPN[0].cmNoise[(i*kFstNumRStripsPerSensor+j)*kFstNumTimeBins+k] / 100.0;
```

Comment:
The c-style cast (float) here has no effect and is not needed. The divisor 100.0 will force the whole calculation to be done in double precision. The final double to float cast to mCmnVec[i][j][k] is implicit.